src/grid_search.py

```diff
@@ -160,100 +160,98 @@ def process_params(tasks, results, G, best_paths, INITIAL_POPULATIONS):
     print(f"Done {os.getpid()}")
 
 
-if __name__ == "__main__":
+def main() -> None:
+    # Setup of the city
+    G, best_paths = generate_city_graph(N)
+
+    # Parameters
+    INITIAL_POPULATIONS = [create_initial_population(G, best_paths) for _ in range(3)]
+
+    float_param_list: list[float] = [0.2, 0.8]
+    lower_bound_params: list[float] = [0.1, 0.3]
+    higher_bound_params: list[float] = [0.7, 0.9]
+
+    grid_search_params = {
+        "survival_functions": [1],  # range(len(SURVIVAL_FUNCTIONS)),
+        "epochs": [100],
+        "chance_rot_cycle": [0.5],
+        "chance_rot_right": [0.5],
+        "chance_invert": [0.5],
+        "chance_erase_stop": lower_bound_params,
+        "chance_add_stop": lower_bound_params,
+        "chance_add_stop_mix": lower_bound_params,
+        "chance_replace_stops": lower_bound_params,
+        "chance_replace_stops_proximity": lower_bound_params,
+        "chance_create_line": [0.1],
+        "chance_cycle": [0.1],
+        "chance_erase_line": [0.25],
+        "chance_merge": [0.25],
+        "chance_merge_mix": float_param_list,
+        "chance_split": [0.75],
+        "cycle_stops_shift": float_param_list,
+        "chance_merge_specimen": [0.5],
+        "chance_cycle_stops_shift": lower_bound_params,
+        "chance_line_based_merge": lower_bound_params,
+    }
+
+    # Setup of the grid search
+    parallel_units = 1
+    cpu_count = os.cpu_count()
+    if cpu_count is not None:
+        parallel_units = cpu_count
+
+    print("Parallel units:", parallel_units)
+
+    # fixed order list
+    # order of keys in params_keys define column order in csv file
+    # must stay the same for all processes
+    params_keys = list(grid_search_params.keys())
+    queue: "Queue[dict]" = Queue()
```

Comment:
It somehow works for me, but WTF is That type? Why not Just `Queue[dict[str, Any]]`?

Reply:
@kosmydel